pkg/controller/pod_volume_backup_controller.go

```diff
@@ -18,330 +18,220 @@ package controller
-		"namespace": req.Namespace,
-		"name":      req.Name,
+// +kubebuilder:rbac:groups=velero.io,resources=podvolumebackup,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=velero.io,resources=podvolumebackup/status,verbs=get;update;patch
+func (r *PodVolumeBackupReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithFields(logrus.Fields{
+		"controller":      "podvolumebackup",
+		"podvolumebackup": req.NamespacedName,
 	})
 
-	if len(req.OwnerReferences) == 1 {
-		log = log.WithField("backup", fmt.Sprintf("%s/%s", req.Namespace, req.OwnerReferences[0].Name))
-	}
-
-	return log
-}
-
-func getOwningBackup(req *velerov1api.PodVolumeBackup) string {
-	return fmt.Sprintf("%s/%s", req.Namespace, req.OwnerReferences[0].Name)
-}
-
-func (c *podVolumeBackupController) processBackup(req *velerov1api.PodVolumeBackup) error {
-	log := loggerForPodVolumeBackup(c.logger, req)
-
-	log.Info("Backup starting")
-
-	var err error
+	pvb := velerov1api.PodVolumeBackup{}
+	if err := r.Client.Get(r.Ctx, req.NamespacedName, &pvb); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Debug("Unable to find PodVolumeBackup")
+			return ctrl.Result{}, nil
+		}
 
-	// update status to InProgress
-	req, err = c.patchPodVolumeBackup(req, func(r *velerov1api.PodVolumeBackup) {
-		r.Status.Phase = velerov1api.PodVolumeBackupPhaseInProgress
-		r.Status.StartTimestamp = &metav1.Time{Time: c.clock.Now()}
-	})
-	if err != nil {
-		log.WithError(err).Error("Error setting PodVolumeBackup StartTimestamp and phase to InProgress")
-		return errors.WithStack(err)
+		return ctrl.Result{}, errors.Wrap(err, "getting PodVolumeBackup")
 	}
 
-	pod, err := c.podLister.Pods(req.Spec.Pod.Namespace).Get(req.Spec.Pod.Name)
-	if err != nil {
-		log.WithError(err).Errorf("Error getting pod %s/%s", req.Spec.Pod.Namespace, req.Spec.Pod.Name)
-		return c.fail(req, errors.Wrap(err, "error getting pod").Error(), log)
+	if len(pvb.OwnerReferences) == 1 {
+		log = log.WithField(
+			"backup",
+			fmt.Sprintf("%s/%s", req.Namespace, pvb.OwnerReferences[0].Name),
+		)
 	}
 
-	volumeDir, err := kube.GetVolumeDirectory(pod, req.Spec.Volume, c.pvcLister, c.pvLister)
+	// Initialize patch helper.
+	patchHelper, err := patch.NewHelper(&pvb, r.Client)
 	if err != nil {
-		log.WithError(err).Error("Error getting volume directory name")
-		return c.fail(req, errors.Wrap(err, "error getting volume directory name").Error(), log)
+		log.WithError(err).Error("error getting a patch helper to update this resource")
+		return ctrl.Result{}, err
 	}
 
-	pathGlob := fmt.Sprintf("/host_pods/%s/volumes/*/%s", string(req.Spec.Pod.UID), volumeDir)
-	log.WithField("pathGlob", pathGlob).Debug("Looking for path matching glob")
+	defer func() {
+		// Attempt to Patch the pvb object and status after each reconciliation.
+		if err := patchHelper.Patch(r.Ctx, &pvb); err != nil {
+			log.WithError(err).Error("error updating download request")
```

Comment:
The error message should've not be "error updating download request".

Reply:
I got rid of patch helper completely and now use the r.Client's Update() method.